docs/dev/table/functions/udfs.md

```diff
@@ -1268,693 +1044,770 @@ class WeightedAvg extends AggregateFunction[JLong, CountAccumulator] {
+ * there is an update, we have to retract old records before sending new updated ones. The emitUpdateWithRetract()
+ * method will be used in preference to the emitValue() method if both methods are defined in the table aggregate
+ * function, because the method is treated to be more efficient than emitValue as it can output
+ * values incrementally.
+ *
+ * param: accumulator           the accumulator which contains the current aggregated results
+ * param: out                   the retractable collector used to output data. Use the collect() method
+ *                              to output(add) records and use retract method to retract(delete)
+ *                              records.
  */
-class Top2Accum {
-  var first: JInteger = _
-  var second: JInteger = _
-  var oldFirst: JInteger = _
-  var oldSecond: JInteger = _
-}
+public void emitUpdateWithRetract(ACC accumulator, RetractableCollector<T> out)
+{% endhighlight %}
+</div>
 
-/**
- * The top2 user-defined table aggregate function.
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+/*
+ * Called every time when an aggregation result should be materialized. The returned value could
+ * be either an early and incomplete result (periodically emitted as data arrives) or the final
+ * result of the aggregation.
+ *
+ * Compared to emitValue(), emitUpdateWithRetract() is used to emit values that have been updated. This method
+ * outputs data incrementally in retraction mode (also known as "update before" and "update after"). Once
+ * there is an update, we have to retract old records before sending new updated ones. The emitUpdateWithRetract()
+ * method will be used in preference to the emitValue() method if both methods are defined in the table aggregate
+ * function, because the method is treated to be more efficient than emitValue as it can output
+ * values incrementally.
+ *
+ * param: accumulator           the accumulator which contains the current aggregated results
+ * param: out                   the retractable collector used to output data. Use the collect() method
+ *                              to output(add) records and use retract method to retract(delete)
+ *                              records.
  */
-class Top2 extends TableAggregateFunction[JTuple2[JInteger, JInteger], Top2Accum] {
-
-  override def createAccumulator(): Top2Accum = {
-    val acc = new Top2Accum
-    acc.first = Int.MinValue
-    acc.second = Int.MinValue
-    acc.oldFirst = Int.MinValue
-    acc.oldSecond = Int.MinValue
-    acc
+def emitUpdateWithRetract(accumulator: ACC, out: RetractableCollector[T]): Unit
+{% endhighlight %}
+</div>
+
+</div>
+
+### Retraction Example
+
+The following example shows how to use the `emitUpdateWithRetract(...)` method to emit only incremental
+updates. In order to do so, the accumulator keeps both the old and new top 2 values.
+
+_Note_: If the N of Top N is big, it might be inefficient to keep both the old and new values. One way to
```

Comment:
```suggestion
If the N of Top N is big, it might be inefficient to keep both the old and new values. One way to
```

I don't think we need the note comment because this is what we're trying to show.